### ISSUE TITLE
Layout/SpaceInsideHashLiteralBraces does not find all violations in some situations and has to be run at least twice

### ISSUE THREAD

**Issue by eric-christian** [rubocop-hq/rubocop#8642]
## Expected behavior

I expect the cop to find all violations:
```bash
❯ rubocop demo.rb
Inspecting 1 file
C

Offenses:

demo.rb:8:3: C: Layout/SpaceInsideHashLiteralBraces: Space inside { missing.
  {key: 1}
  ^
demo.rb:8:10: C: Layout/SpaceInsideHashLiteralBraces: Space inside } missing.
  {key: 1}
         ^

1 file inspected, 2 offenses detected, 2 offenses auto-correctable
```

## Actual behavior

It only finds one violation:
```bash
❯ rubocop demo.rb
Inspecting 1 file
C

Offenses:
demo.rb:8:3: C: Layout/SpaceInsideHashLiteralBraces: Space inside { missing.
  {key: 1}
  ^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## Steps to reproduce the problem

Create the file `demo.rb`:
```ruby
# frozen_string_literal: true

def abc
  {}
end

def xyz
  {key: 1}
end
```

Run rubocop:
```bash
❯ rubocop demo.rb
```

I don't know why, but the method body of `abc` is relevant. If you change it from `{}` to for example `'a string'`, then the cop works as expected.

## RuboCop version

```bash
❯ rubocop -V
0.90.0 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.7.1 x86_64-darwin19)
```
